source/material_model/rheology/strain_dependent.cc

```diff
@@ -450,6 +469,41 @@ namespace aspect
 
       }
 
+      template <int dim>
+      std::array<double, 3>
+      StrainDependent<dim>::
+      apply_temperature_dependence_to_strain_weakening_factors(const unsigned int j,
+                                                               const double temperature,
+                                                               std::array<double, 3> weakening_factors) const
+      {
+
+        //
+        //  ------            -------- 1
+        //        \          /
+        //         \        /
+        //          \______/ _ _ _ _ _ viscous_weakening_factor[2]
+        //
+        //  ----------------------------> T
+        //      T0  T1   T2  T3
+
```

Comment:
With the change to the input parameter you have to do this:
```
std::array<double, 3> output = weakening_factors;
```

This is not slower then what happens at the moment, because handing over the array by value also creates a copy.

Reply:
I hope I understood what you meant